WalletWasabi.Tests/Helpers/HttpClientWrapper.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Tor.Http;
+
+namespace WalletWasabi.Tests.Helpers
+{
+	public class HttpClientWrapper : IHttpClient
```

Comment:
This is a flawed class hacks around a flawed concept introduced previously, which is that we created a `IHttpClient` interface knowing well `HttpClient` doesn't implement it. Sad, but cACK.

Reply:
what about adding a `IHttpSingleEndpointClient` (or whatever) that always has a base URI and only supports requests with a relative URI and no `Host` header (i.e. same as `IHttpClient` except with a non nullable base), and eventually (will take a bit of refactoring) make `IHttpClient` really match `HttpClient`?

Reply:
No, this is _the only way_ I've found to work around a flawed concept introduced previously.

Reply:
what i mean is that the previously introduced concept really is flawed, but it could be removed eventually if the relative URI functionality is separated from it

Reply:
Specifically the flaw is that relative URIs and absolute URIs are conceptually distinct, but can be represented with the same type  with the same type (one way is consequence of HTTP 1.0 only having only the relative concept and HTTP 1.1 adding the `Host` header for virtual hosts, but this is further complicated by the fact that `HttpRequestMessage` also supports absolute URLs and transforms them like an HTTP proxy would).

A narrower API that *only* admits requests with the host and base path implicit can guarantees about the transport layer, which is important for clarity of the tor circuit isolation, so arguably it should be separated from `IHttpClient`, which is general purpose instead of implementing both behaviours, which have very different use cases, in the same API.

Reply:
I have no idea how to make this better but keep in mind this is only for tests.